### ISSUE TITLE
Playlist Manager search error

### ISSUE THREAD

**Issue by irodimus** [blacktwin/JBOPS#200]
With the latest commit 65b424d3cc9e1da95add9c1e270cc17b269b2b20 for playlist_manager.py, using the search argument fails and errors. An example I use is:
```
plex-ht:Scripts administrator$ python playlist_manager.py --jbop random --action update --name "41 South Park" --libraries "TV Shows" --limit 50 --search grandparentTitle="south park"
Traceback (most recent call last):
  File "playlist_manager.py", line 828, in <module>
    keys_list = build_playlist(opts.jbop, libraries, opts.days, opts.top, filters, search, opts.limit)
  File "playlist_manager.py", line 512, in build_playlist
    keys_list = get_content(libraries, jbop, filters, search, limit)
  File "playlist_manager.py", line 413, in get_content
    search_lst = multi_filter_search(keywords, plex_library, search_eps=True)
  File "playlist_manager.py", line 344, in multi_filter_search
    for episode in library.searchEpisodes(**{key: values}):
  File "/usr/local/lib/python2.7/site-packages/plexapi/library.py", line 593, in searchEpisodes
    return self.search(libtype='episode', **kwargs)
  File "/usr/local/lib/python2.7/site-packages/plexapi/library.py", line 493, in search
    args[category] = self._cleanSearchFilter(category, value, libtype)
  File "/usr/local/lib/python2.7/site-packages/plexapi/library.py", line 514, in _cleanSearchFilter
    raise BadRequest('Unknown filter category: %s' % category)
plexapi.exceptions.BadRequest: Unknown filter category: grandparentTitle__icontains
```
Changing the metadata filter to anything else gives this error except for the Unknown filter category result with the current metadata filter being used.

**Comment by blacktwin** [comment]
Hmmm, and it works fine before that commit?

**Comment by irodimus** [comment]
To be honest, I had not noticed till recently since the process is automated, but it had been working without issue previously. I'll see if I can find an older commit to see if it works there.

**Comment by blacktwin** [comment]
Seems to work fine when you use the correct command.

`playlist_manager.py --jbop random --action update --name "41 South Park" --libraries "TV Shows" --limit 50 --filter show.title="south park"`

`grandparentTitle` is not the correct searchable name, `show.title` is. Also you're using the `--search` instead of `--filter`. Filter mirrors Plex's filtering mechanic where search will attempt to find the keyword in the searchable field provided. So `--filter show.title="south park"` will find any show titles named "South Park" where `--search show.title="south park"` will look for "south park" in the show's title. So technically it **should** work both ways. There is still in issue to be resolved in the code but for now change your command string to use filter instead.

**Comment by blacktwin** [comment]
Ok I've reverted that commit. I'll keep this issue open until I can issue the search issue. Please let me know that the `--filter` change in your command works and resolves your issue. Some it maybe linked to https://github.com/pkkid/python-plexapi/issues/392.

**Comment by irodimus** [comment]
I tried
```
playlist_manager.py --jbop random --action update --name "41 South Park" --libraries "TV Shows" --limit 50 --filter show.title="south park"
```
but I got this result
```
(show.title) was not found in filters list: [network, unwatched, country, contentRating, collection, label, director, duplicate, studio, actor, year, genre, guid, resolution, decade, mood]
```

I changed the playlist manager command after getting the latest commit to
```
playlist_manager.py --jbop random --action update --name "41 South Park" --libraries "TV Shows" --limit 50 --search show.title="south park"
```
and it appeared to complete but didn't create a playlist
```
Deleting the playlist(s)...
Creating playlist(s)...
Done.
```
So I reverted back to this command with the latest commit
```
python playlist_manager.py --jbop random --action update --name "41 South Park" --libraries "TV Shows" --limit 50 --search grandparentTitle="south park"
```
And that works now.

**Comment by blacktwin** [comment]
So the first error could be that you don't have the latest version of the PlexAPI-Python. 

`pip install git+https://github.com/pkkid/python-plexapi`
This will pull from github instead of pypi. Updating from github will pull the latest version and that will have the updated filter list. You can now use `--filter show.title "Whatever"` or `--search grandparentTitle "Whatever"` and get similar results.

So in summary, `--filter` mimics Plex's own filtering found in the WebUI and thus uses the same categories found in the `APPROVED_FILTERS`. `--search` actually searches for any metadata field, so grandparentTitle, parentTitle, title, summary, etc. basically any video object's attribute.

**Comment by irodimus** [comment]
I went ahead and did the update to PlexAPI. I wasn't aware I was running an old version, 3.0.4. I'll try this when I get a few minutes. Also, is the "=" not required when doing a --filter or --search?

**Comment by blacktwin** [comment]
> Also, is the "=" not required when doing a --filter or --search?

It is, sorry. Edited the previous comment.

**Comment by blacktwin** [comment]
@irodimus are you still having issues with this script? Sorry it's been awhile. Just want to see if I need to do to close this.

**Comment by irodimus** [comment]
Yes, this issue has been fixed and can be closed.
